Wii U / Video: Fixed resume of video afrer the home menu close

Wohlstand opened the pull request:
This bug happens just when V-Sync is off and attempting to open and close the home menu. To fix the black/frozen screen it's just enough to call the `GX2SetSwapInterval(0);` right after entering the foreground back.

P.S. I checked on the WUT side, and seems, such workaround is required since it's a bug of the Wii U firmware itself.

## Description

Demo of buggy behaviour at [my project](https://github.com/TheXTech/TheXTech/):

https://github.com/user-attachments/assets/1e4ee887-b16c-4c35-90de-9d97fa485982

Demo of the fixed behaviour at my project:

https://github.com/user-attachments/assets/889f8510-17f2-452d-8f93-905657cbc076

Unexpectedly, this bug affects the EasyRPG (ping @Ghabry to use the newer SDL2 for Wii U after this fix arrive) too when disabling V-Sync, and re-enabling V-Sync restores back the screen:

https://github.com/user-attachments/assets/9e621ef1-8472-4a0d-9615-a04f26338320

Wohlstand commented:
@GaryOderNichts, can you remind me what's left to test/ensure until merge this?

GaryOderNichts commented:
Were there any issues with doing it like this https://github.com/devkitPro/SDL/commit/ad29a9f2ea5fd157103fe9e462a2635950f56ee6? Since this is a workaround only necessary for swap intervals of 0, it might be the simpler solution. I would prefer moving it into `WIIU_ForegroundReleased` though.

Wohlstand commented:
> Were there any issues with doing it like this [ad29a9f](https://github.com/devkitPro/SDL/commit/ad29a9f2ea5fd157103fe9e462a2635950f56ee6)? Since this is a workaround only necessary for swap intervals of 0, it might be the simpler solution. I would prefer moving it into `WIIU_ForegroundReleased` though.

My workaround that I did, is supposed to perform save/restore, and it just worked. I could try the thing you suggested, and if it will work, then flawless, but if it fails, then just keep my initial workaround.

Wohlstand committed:
Wii U / Video: Fixed resume of video afrer the home menu close

This bug happens just when V-Sync is off and attempting to open and close the home menu. To fix the black/frozen screen it's just enough to call the `GX2SetSwapInterval(0);` right after entering the foreground back.

Wohlstand commented:
> Were there any issues with doing it like this [ad29a9f](https://github.com/devkitPro/SDL/commit/ad29a9f2ea5fd157103fe9e462a2635950f56ee6)? Since this is a workaround only necessary for swap intervals of 0, it might be the simpler solution. I would prefer moving it into `WIIU_ForegroundReleased` though.

Okay, I done the test, and the small thing seems just fine! I applied it to the pull-request right now.

The test of the tweak that you sent to me:

https://github.com/user-attachments/assets/e02352ec-f043-49fb-9aa0-2536a28fa939

The control test over clear `wiiu-sdl2-2.28` branch without this change (the bug is here):

https://github.com/user-attachments/assets/3a291690-4675-4084-98df-b2233161295e

GaryOderNichts commented:
Can you move the workaround into `WIIU_ForegroundReleased` and out of the handle ProcUI check? Also replace the comment with something like `Workaround for leaving foreground with a swap interval 0`.

Wohlstand committed:
Move out of handleProcUI cond and tweaked comment

GaryOderNichts commented:
> I tested, if I move to `WIIU_ForegroundReleased`, it stops to work at all, it just **does not works**. I checked at multiple places:

My bad, I meant `WIIU_ForegroundAcquired`. LGTM.

Wohlstand commented:
> LGTM

What do you mean?

Just a note: foreigners like me aren't familiar with slang acronyms.

GaryOderNichts commented:
> > LGTM
> 
> What do you mean?
> 
> Just a note: foreigners like me aren't familiar with slang acronyms.

It's a common term used to approve code reviews (https://en.wiktionary.org/wiki/LGTM).

Wohlstand commented:
> > > LGTM
> > 
> > 
> > What do you mean?
> > Just a note: foreigners like me aren't familiar with slang acronyms.
> 
> It's a common term used to approve code reviews (https://en.wiktionary.org/wiki/LGTM).

I see, I didn't knew. And I didn't even saw usage of this phrase around, I am on Github for more than 10 years and I contributed lots of projects as well as I develop my own, but nobody used that acronym across topics that I read.

Also, thank you very much for taking my tweak!
